### ISSUE TITLE
Index spinner goes forever and doesn't show new files, only modified ones

### ISSUE THREAD

**Issue by jeremyk** [gitx/gitx#86]
Happy to add more detail but not really sure how to debug this. I have an old Rowanj fork running that works fine but have the issue with this (and the older scp fork). I wonder if something just got corrupted for that repo but not sure how to clear out the data.

**Comment by tiennou** [comment]
In the stage view ?

**Comment by jeremyk** [comment]
Yes

**Comment by tiennou** [comment]
I'd really like "more detail" please :
- Can you specify the repo this happens on ?
- Does `git status` report anything weird ? 
- Does `git fsck` report anything (that's in case of corruption, as you suspect) ?
- What kind of files are "new", big binary ones, lots of source-file-type ones ?

**Comment by jeremyk** [comment]
Thanks for looking. Sorry I am not more helpful and wish I could share the repo but it is private.

Git status is normal and shows the untracked files. Yes they are a bunch random files (some in a bin folder). I can move the repo down a level and get rid of those but wanted to understand what was going on first.

Git fsck shows a bunch of dangling commits and blobs but nothing too odd:
```
dangling commit c40c7421c9f84dcf176a490fbee47175663607a3
dangling commit f51380ad2c768690feceee47af611c2f5add32a3
dangling blob 2214ac412049b43e08b9d4e47db67bf5eb9ae5cd
dangling blob 2539d8832e3172d480ea01dd0e761b6bb034df85
...
```

Here is a screenshot in case my description wasn't clear. Old Rowanj on the right:
<img width="1190" alt="screen shot 2017-04-25 at 8 53 54 am" src="https://cloud.githubusercontent.com/assets/223474/25395255/519fbb4c-2995-11e7-81a1-9b670a7c924f.png">

**Comment by claytonrcarter** [comment]
I'm having the same issue.  Stage view shows tracked files fine, both staged and unstaged, but doesn't seem to show new/untracked files.  Then there's the index spinner...

`git status` looks normal, `git fsck` showed lots (100+) of dangling blobs and commits, and I have lots (many hundreds, maybe 1000+) of plain source files in my worktree that need to be staged & commited.

**Comment by claytonrcarter** [comment]
After staging all of the files manually, I opened gitx again and found that it's now correctly showing untracked files (there are only 2) in the "Unstaged Changes" pane, but it's showing nothing in the "Staged Changes" pane, even though there are ~10 tracked+staged and many, many more untracked+staged files.  Before manually staging the files, this showed the tracked files that I had staged.  Now it's just empty.  I have 1649 files staged, 1636 of them are new files.

After committing these changes (manually) and reopening gitx, everything goes back to normal and the index spinner disappears.
